app/src/main/java/com/dgaf/happyhour/Device.java

```diff
@@ -0,0 +1,24 @@
+package com.dgaf.happyhour;
+
+import android.content.Context;
+import android.telephony.TelephonyManager;
+
+import java.util.UUID;
+
+/**
+ * Created by Adam on 8/13/2015.
+ */
+public class Device {
+    public static String getDeviceId(Context context) {
+        final TelephonyManager tm = (TelephonyManager) context.getSystemService(Context.TELEPHONY_SERVICE);
+
```

Comment:
Are you allowed to assign a `String` after declaring it `final`?  Or am I getting mixed up with C++